sql/core/src/main/scala/org/apache/spark/sql/classic/Catalog.scala

```diff
@@ -184,25 +185,19 @@ class Catalog(sparkSession: SparkSession) extends catalog.Catalog with Logging {
     try {
       Some(makeTable(nameParts))
     } catch {
-      case e: AnalysisException if e.getCondition == "TABLE_OR_VIEW_NOT_FOUND" => None
-      // Swallow non-fatal throwables when resolving a table or view and
-      // return a table or view with partial results to
-      // prevent listTables from breaking easily due to a broken table or view.
-      case NonFatal(e) if !isTempView =>
```

Comment:
The change seems good but shouldn't we still keep this though? Other exceptions might happen that doesn't inherit `SparkThrowable`.

Reply:
There are risks of working with partial data unaware that there are broken tables.

Reply:
👌